API/Card.cs

```diff
@@ -0,0 +1,12 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace API
+{
+    public abstract class Card
+    {
+        public abstract string name { get; }
```

Comment:
Maybe we could make name an `enum` instead of string? This way we wouldn't need to hard code type values inside `CardCreator`